.github/workflows/cmake_build.yml

```diff
@@ -0,0 +1,259 @@
+name: Build and Test on Push
+on:
+  push:
+  pull_request:
+    branches: [master]
+
+jobs:
+  # These test the 'standard' build on several systems with gcc + clang.
+  standard_tests:
+    name: ${{ matrix.os }} & ${{ matrix.compiler.CC }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, 	ubuntu-20.04, macos-10.15]
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+
+    steps:
+      - name: Get Source
+        uses: actions/checkout@v2
+  
+      - name: "Linux: Run Apt-Get"
+        if: startsWith(matrix.os, 'ubuntu') && matrix.os != 'ubuntu-20.04'
+        run: sudo apt-get install elfutils libzstd1-dev
+  
+      - name: "Linux (Ubuntu 20.04): Run Apt-Get"
+        if: matrix.os == 'ubuntu-20.04'
+        run: sudo apt-get install elfutils libzstd-dev
+  
+      - name: "Setup common ENV"
+        uses: allenevans/set-env@v1.1.0
+        with:
+          CC: ${{ matrix.compiler.CC }}
+          CXX: ${{ matrix.compiler.CXX }}
+          VERBOSE: 1
+          ENABLE_CACHE_CLEANUP_TESTS: true
+          CTEST_OUTPUT_ON_FAILURE: ON
```

Comment:
Would it be possible to set the environment variables on the job level using the [common GitHub actions way](https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#about-environment-variables), thereby avoiding using allenevans/set-env and the resulting "Unexpected input(s)" warnings?

Reply:
Yes, but the trouble is one would need to set them at every single step. Maybe that's not an issue if we end up with 1-2 steps. But at the moment I have those build + build binary + build source steps which all need the same env variables. I'll revisit after fixing the rest.

Reply:
Hm, not sure I understand. It's possible to set environment variables on the **job** level so they don't have to be set for every single **step**. Like this: https://github.com/jrosdahl/ccache/commit/fc43c6fef32d6a47ed27920765983193f73a164b. CI run: https://github.com/jrosdahl/ccache/runs/964268833. Am I missing something?

Reply:
For some reason I was under the impression it's not possible. In current setup setting those on Step level seems fine. Maybe in future cleanups / improvements it will make sense to move them.